src/test/test-tpm2.c

```diff
@@ -547,6 +548,107 @@ TEST(tpm2_get_primary_template) {
         }
 }
 
+#define SHA256_T0 "0000000000000000000000000000000000000000000000000000000000000000"
+#define SHA256_T1 "17b7703d9d00776310ba032e88c1a8c2a9c630ebdd799db622f6631530789175"
+#define SHA256_T2 "12998c017066eb0d2a70b94e6ed3192985855ce390f321bbdb832022888bd251"
+#define SHA256_T3 "c3a65887fedd3fb4f5d0047e906dff830bcbd1293160909eb4b05f485e7387ad"
+#define SHA256_T4 "6491fb4bc08fc0b2ef47fc63db57e249917885e69d8c0d99667df83a59107a33"
+
+#define DIGEST_COMPARE(d1, d2)                                          \
+        ({                                                              \
+                assert_se((d1).size == (d2).size);                      \
+                assert_se(memcmp_safe((d1).buffer, (d2).buffer, (d1).size) == 0); \
+        })
```

Comment:
why a macro for this? sounds like it could as well be a function? what am i missing?

Reply:
i actually should have removed this, i would up not using it; i'll change the other macros to functions though